[DOCS] Adds data frame analytics field type and document limitation (#501)
elastic/stack-docs: docs/en/stack/ml/df-analytics/dfanalytics-limitations.asciidoc

If there are missing values in feature fields (fields that are subjects of the 
{dfanalytics}), then the document that contains the fields with the missing 
values will be skipped during the analysis.

[float]
[[dfa-field-type-docs-limitations]]

> === {dfanalytics-cap} field type and document limitation

Reviewer: I'm not sure I understand why we need this section. Are there lots of other type/document limitations expected here? If not, I think it should be removed and the lower-level sections should be bumped up.

[float]
==== {oldetection-cap}

{oldetection-cap} requires numeric or boolean data to analyze. The algorithms 
don't support missing values (see also <<dfa-missing-fields-limitations>>), 

> therefore fields other than numeric or boolean will be ignored. Documents where 

Reviewer: Minor suggestion:
```suggestion
therefore fields that have data types other than numeric or boolean are ignored. Documents where 
```

> included fields contain missing values, null values, or an array also will be 

Reviewer: ```suggestion
included fields contain missing values, null values, or an array are also
```

> ignored. Therefore a `dest` index may contain documents that don't have an 

Reviewer: We don't refer to the "dest" keyword anywhere else in this page, so I think it's better to use the full name...
```suggestion
ignored. Therefore a destination index may contain documents that don't have an 
```

> {olscore}. These documents are still reindexed from `source` to `dest`, but they 

Reviewer: Ditto..
```suggestion
{olscore}. These documents are still reindexed from the source index to the destination index, but they 
```

are not included in the {oldetection} analysis and therefore no {olscore} is 
computed.

[float]
==== {regression-cap}

{regression-cap} supports fields that are numeric, boolean, text, keyword and ip 

> and is tolerant of missing values. Fields that are supported will be included in 

Reviewer: Minor suggestion:
```suggestion
. It is also tolerant of missing values. Fields that are supported are included in 
```

> the analysis, other fields will be ignored. Documents where included fields 

Reviewer: ```suggestion
the analysis, other fields are ignored. Documents where included fields 
```

> contain an array will also be ignored. Documents in `dest` that don't contain a 

Reviewer: ```suggestion
contain an array are also ignored. Documents in the destination index that don't contain a 
```

> results field will not be included in the {reganalysis}.

Reviewer: ```suggestion
results field are not included in the {reganalysis}.
```
